modules/assim.sequential/R/sda.enkf_MultiSite.R

```diff
@@ -355,17 +355,33 @@ sda.enkf.multisite <- function(settings,
       #reformatting params
       new.params <- sda_matchparam(settings, ensemble.samples, site.ids, nens)
     }
-      #sample met ensemble members
-      #TODO: incorporate Phyllis's restart work
-      #      sample all inputs specified in the settings$ensemble not just met
-      inputs <- PEcAn.settings::papply(conf.settings,function(setting) {
-        PEcAn.uncertainty::input.ens.gen(
-          settings = setting,
-          input = "met",
-          method = setting$ensemble$samplingspace$met$method,
-          parent_ids = NULL 
-        )
-       })
+      
+   #sample all inputs specified in the settings$ensemble
+   #now looking into the xml
+   samp <- conf.settings$ensemble$samplingspace
+   #finding who has a parent
+   parents <- lapply(samp,'[[', 'parent')
+   #order parents based on the need of who has to be first
+   order <- names(samp)[lapply(parents, function(tr) which(names(samp) %in% tr)) %>% unlist()] 
+   #new ordered sampling space
+   samp.ordered <- samp[c(order, names(samp)[!(names(samp) %in% order)])]
+   #performing the sampling
+   inputs <- vector("list", length(conf.settings))
+   # For the tags specified in the xml I do the sampling
+   for (s in seq_along(conf.settings)){
+      if (is.null(inputs[[s]])) {
+        inputs[[s]] <- list() 
+      }
+     for (i in seq_along(samp.ordered)){
+         #call the function responsible for generating the ensemble
+        inputs[[s]][[names(samp.ordered)[i]]] <- input.ens.gen(settings=conf.settings[[s]],
```

Comment:
is this sampling occurring for each site independently or across all sites simultanously?

Reply:
For each site independently.

Reply:
Is independent sampling (which is NOT what we want) something new here or are you just continuing the status quo. If it's the status quo I can see punting the fix to a new PR, but if you moved from joint to independent sampling you'll need to revert it back to joint.

Reply:
No, what I mean is that if model ensemble member 1 at site 1 samples met ensemble k (e.g. 6) then model ensemble member 1 needs to also use met ensemble member k at sites 2 through 6400. Even though the specific met data is different from site to site, the spatiotemporal values within a specific met ensemble member share a covariance structure.  Similarly if ensemble member 1 at site 1 samples a particular parameter vector with a specific row number, then all other sites need to use the same posterior row number (though row k for one PFT will obviously point to a different set of parameters than row k of another PFT). By having the same parameters, met, etc across sites we induce the correct spatiotemporal covariance structure across sites.